components/org.wso2.carbon.identity.provisioning.choreo.connector.scim/src/main/java/org/wso2/carbon/identity/provisioning/choreo/connector/scim/SCIMProvisioningChoreoConnector.java

```diff
@@ -0,0 +1,332 @@
+            user.setUserName(userName);
+
+            try {
+
+                String contentType = this.scimProvider.getProperty("Content-Type");
+                if (contentType == null) {
+                    contentType = SCIMChoreoConfigConstants.ELEMENT_NAME_JSON_TYPE;
+                }
+                assert userName != null;
+                String slashRemovedUserName = userName.replace('/', ',');
+
+                String endPointUrl = scimProvider.getProperty("apiEndpoint") + "/" + slashRemovedUserName;
+                String tokenValue = scimProvider.getProperty("apiToken");
+                String authorizationHeaderValue = SCIMChoreoConfigConstants.ELEMENT_NAME_TOKEN_TYPE + tokenValue;
+
+                HttpDelete delete = new HttpDelete(endPointUrl);
+                delete.setHeader("Accept", SCIMChoreoConfigConstants.ELEMENT_NAME_ACCEPT_TYPE);
+                delete.setHeader("Content-type", contentType);
+                delete.setHeader("Authorization", authorizationHeaderValue);
+
+                try (CloseableHttpClient httpClient = HttpClients.createDefault();
+                     CloseableHttpResponse response = httpClient.execute(delete)) { }
+
+            } catch (IOException e) {
+                throw new IdentityProvisioningException("Error while sending the user information to the API", e);
+            }
+
+        } catch (Exception e) {
+            throw new IdentityProvisioningException("Error while deleting user.", e);
+        }
+    }
+
+    private void updateUser(ProvisioningEntity userEntity, ProvisioningOperation provisioningOperation) throws
+            IdentityProvisioningException {
+        try {
+
+            List<String> userNames = getUserNames(userEntity.getAttributes());
+            String userName = null;
+
+            if (CollectionUtils.isNotEmpty(userNames)) {
+                userName = userNames.get(0);
+            }
+            User user;
+            // Get single-valued claims.
+            Map<String, String> singleValued = getSingleValuedClaims(userEntity.getAttributes());
+            // If user created through management console, claim values are not present.
+            user = (User) AttributeMapper.constructSCIMObjectFromAttributes(singleValued,
+                    SCIMConstants.USER_INT);
+            user.setSchemaList(Arrays.asList(SCIMConstants.CORE_SCHEMA_URI));
+            user.setUserName(userName);
+            setUserPassword(user, userEntity);
+
+            // Adding third party client.
+            try {
+
+                String contentType = this.scimProvider.getProperty("Content-Type");
+                if (contentType == null) {
+                    contentType = SCIMChoreoConfigConstants.ELEMENT_NAME_JSON_TYPE;
+                }
+                String endPointUrl = scimProvider.getProperty("apiEndpoint") + "/";
+                String tokenValue = scimProvider.getProperty("apiToken");
```

Comment:
Shall we define these properties as a constant?

Reply:
Sure, I will add